File: backend/internal/database/models/storyboard.model.go
package models

type StoryboardEntity struct {
	BaseItem
	// CampaignEntity body copy
	BodyCopy string `dynamodbav:"bodyCopy"`
	// Call to action text
	Cta string `dynamodbav:"cta"`
	// CampaignEntity headline
	Headline string `dynamodbav:"headline"`
	// Associated job ID
	JobID string `dynamodbav:"jobId"`
	// StoryboardEntity slides
	Slides []SlideEntity `dynamodbav:"slides"`
}

File: frontend/src/lib/models/storyboard.model.ts
import type { ICampaign } from "./campaign.model";
import type { ISlide } from "./slide.model";

export interface IStoryboard {
  id: string;
  /**
   * ICampaign body copy
   */
  bodyCopy: string;
  /**
   * Call to action text
   */
  cta: string;
  /**
   * ICampaign headline
   */
  headline: string;
  /**
   * Associated job ID
   */
  jobId: string;
  /**
   * IStoryboard slides
   */
  slides: ISlide[];
}
